src/Compilers/CSharp/Portable/CSharpResources.resx

```diff
@@ -6986,4 +6986,10 @@ To remove the warning, you can use /reference instead (set the Embed Interop Typ
   <data name="ERR_LineContainsDifferentWhitespace" xml:space="preserve">
     <value>Line contains different whitespace than the closing line of the raw string literal: '{0}' versus '{1}'</value>
   </data>
+  <data name="ERR_NoEnumConstraint" xml:space="preserve">
+    <value>Keyword 'enum' cannot be used as a constraint. Did you mean 'System.Enum'?</value>
```

Comment:
```suggestion
    <value>Keyword 'enum' cannot be used as a constraint. Did you mean 'where T : struct, System.Enum'?</value>
```

Reply:
I lean against this change (the original text is a better match to the variety of scenarios supported)

Reply:
I genuinely disagree.  The original text isn't really the recommendation (it would allow System.Enum itself, which most people do not want, and can lead to confusing scenarios since it doesn't actually contrain just to actual enums).  Since we know their intent (when they write `enum`) i think we should suggest the best lang consturct to match that intent.

Reply:
I agree with Cyrus here. It feels like the `struct, System.Enum` version is the intent being expressed by the user.